modules/Administration/Search/MVC/View.php

```diff
@@ -0,0 +1,172 @@
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+/**
+ * Created by PhpStorm.
+ * User: viocolano
+ * Date: 01/08/18
+ * Time: 15:45
+ */
+
+namespace SuiteCRM\Modules\Administration\Search\MVC;
+
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
+
+use Sugar_Smarty;
+use SuiteCRM\Search\SearchWrapper;
+
+require_once __DIR__ . '/../../../Home/UnifiedSearchAdvanced.php';
+
+abstract class View
+{
+    protected $smarty;
+    protected $file;
+
+    /**
+     * View constructor.
+     * @param $file
+     */
+    public function __construct($file)
+    {
+        $this->smarty = new Sugar_Smarty();
+        $this->file = $file;
+    }
+
+    public function preDisplay()
+    {
+        global $mod_strings;
+        global $app_list_strings;
+        global $app_strings;
+        global $sugar_config;
+
+        $errors = array();
+        $this->smarty->assign('MOD', $mod_strings);
+        $this->smarty->assign('APP', $app_strings);
+        $this->smarty->assign('APP_LIST', $app_list_strings);
+        $this->smarty->assign('LANGUAGES', get_languages());
+        $this->smarty->assign('JAVASCRIPT', get_set_focus_js());
+        $this->smarty->assign('error', $errors);
+        $this->smarty->assign('BUTTONS', $this->getButtons());
+
+        $this->smarty->assign('config', $sugar_config['search']);
+    }
+
+    /**
+     * @return string
+     */
+    private function getButtons()
+    {
+        global $mod_strings;
+        global $app_strings;
+
+        return <<<EOQ
+    <input title="{$app_strings['LBL_SAVE_BUTTON_TITLE']}"
+        accessKey="{$app_strings['LBL_SAVE_BUTTON_KEY']}"
+        class="button primary"
+        type="submit"
+        name="save"
+        onclick="return check_form('ConfigureSettings');"
+        value="{$app_strings['LBL_SAVE_BUTTON_LABEL']}" >&nbsp;
+    <input title="{$mod_strings['LBL_CANCEL_BUTTON_TITLE']}" 
+        onclick="document.location.href='index.php?module=Administration&action=index'"
+        class="button"
+        type="button"
+        name="cancel"
+        value="{$app_strings['LBL_CANCEL_BUTTON_LABEL']}" >
+EOQ;
+    }
+
+    public function display()
+    {
+        $this->smarty->display($this->file);
+    }
+
+    /**
+     * @return Sugar_Smarty
+     */
+    public function getSmarty()
+    {
+        return $this->smarty;
+    }
+
+    protected function getEngines()
+    {
+        $engines = [];
+
+        foreach (SearchWrapper::getEngines() as $engine) {
+            $engines[$engine] = translate('LBL_' . $this->from_camel_case($engine));
+        }
+
+        return $engines;
+    }
+
+    protected function from_camel_case($input, $uppercase = true)
+    {
+        preg_match_all('!([A-Z][A-Z0-9]*(?=$|[A-Z][a-z0-9])|[A-Za-z][a-z0-9]+)!', $input, $matches);
```

Comment:
I dont understand this regex. can you please leave a bit explanation or can it be simplier?